systemtest/loadgen/eventhandler/handler.go

```diff
@@ -259,73 +259,107 @@ func New(config Config) (*Handler, error) {
 	return &h, nil
 }
 
+// SendBatchesInLoop will send events in loop, such that it can be used to warm up the remote APM Server,
+// by sending events until the context is cancelled or done channel is closed.
+func (h *Handler) SendBatchesInLoop(ctx context.Context) error {
+	// state is created here because the total number of events in h.batches can be smaller than the burst
+	// and it can lead the sendBatches to finish its cycle without sending the desired burst number of events.
+	// If we keep the state within sendBatches, it will wait for the interval t whenever starting the function as
+	// `s.sent` is set to 0 so it needs to know the previous run's state
+
+	s := state{
+		burst: h.config.Limiter.Burst(),
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			if _, err := h.sendBatches(ctx, &s); err != nil {
+				return err
+			}
+			// reset after batches so it doesn't exceed th limit
+			// but keep the remainder so the next batches know where to start
+			if s.burst > 0 {
+				s.sent = s.sent % s.burst
+			}
```

Comment:
Does it matter if s.sent > s.burst? It was intentional that s.sent be the total events sent, not just the number of events sent for the current iteration.

Reply:
Oh, it's just a safeguard not to exceed the `math.MaxInt`

Reply:
Will revisit it when it becomes problematic.

Reply:
Updated the comment to explain what it does more clearly